portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -136,10 +215,41 @@ private PreparedQuery getAllComments() {
     return datastore.prepare(query);
   }
 
+  /** 
+    * Returns all user comments that contain `search` in either
+    * the comment text or author name. 
+    * @param results All available comment entities in the database.
+    * @param comments An empty list to be filled. 
+    * @param search A string to filter the comments by. 
+    */
+  private ArrayList<Comment> getFilteredComments(PreparedQuery results, 
+    ArrayList<Comment> comments, String search) {
+    results.asIterable().forEach(commentEntity -> {
```

Comment:
actually, this looks like comments will contain All comments. Instead, I would suggest the following method signature. 
getFilteredComments(ArrayList<Comment> comments, String search). 


Converting from PreparedQuery results to List<Comments> should happen elsewhere - Single Responsibility Principle.

Reply:
Yes this makes much more sense! This method signature happens to be a vestigial structure of a previous implementation I had tried (I promise I wouldn't have implemented it like this on purpose)! Fixed :)